File: backend/app/schemas/workspace.py
from uuid import UUID
from typing import Optional
from pydantic import BaseModel

# Dữ liệu dùng để tạo mới
class WorkspaceCreate(BaseModel):
    name: str

# Dữ liệu dùng để cập nhật
class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None

# Dữ liệu trả về cho Client
class WorkspaceRead(BaseModel):
    id: UUID
    name: str
    owner_id: UUID

    class Config:
        from_attributes = True

class InvitationCreate(BaseModel):
    invitee_email: str
    role: str = "Member"

class InvitationRead(BaseModel):
    id: UUID
    workspace_id: UUID
    invitee_email: str
    role: str
    status: str

class InvitationWithWorkspaceRead(BaseModel):
    id: UUID
    workspace_id: UUID
    workspace_name: str  
    inviter_id: UUID
    role: str
    status: str

    class Config:
        from_attributes = True

File: frontend/src/types/workspace.ts
// export interface Workspace {
//   id: string;
//   name: string;
//   owner_id?: string;
// }
export interface Workspace {
  id: string;      // Mapping từ UUID
  name: string;
  owner_id: string;
}

export interface WorkspaceCreate {
  name: string;
}

export interface WorkspaceUpdate {
  name?: string;   // Optional giống như trong Pydantic
}
export interface InvitationCreate {
  invitee_email: string;
  role?: string; // Mặc định là "Member"
}

export interface Invitation {
  id: string;
  workspace_id: string;
  invitee_email: string;
  role: string;
  status: string; // "pending", "accepted", "declined"
}

export interface InvitationWithWorkspace extends Invitation {
  workspace_name: string;
  inviter_id: string;
}
export interface InvitationActionResponse {
  message: string;
  status: string;
}